apps/agent-tars/src/renderer/src/components/LeftSidebar/Settings/AddServerModal.tsx

```diff
@@ -28,114 +28,109 @@ interface AddServerModalProps {
 
 export function AddServerModal({
   isOpen,
   onClose,
   onSubmit,
   initialData,
+  mcpServerNames,
 }: AddServerModalProps) {
   const [isLoading, setIsLoading] = useState(false);
   const mode = initialData?.id ? 'edit' : 'create';
-  const [errors, setErrors] = useState<Record<string, string>>({});
-  const [serverType, setServerType] = useState<MCPServerSetting['type']>(
-    initialData?.type || 'stdio',
-  );
-  const [status, setStatus] = useState<MCPServerSetting['status']>(
-    initialData?.status || 'activate',
-  );
-
-  console.log('mode', mode, 'initialData', initialData);
-
-  const validateForm = (formData: FormData): boolean => {
-    const newErrors: Record<string, string> = {};
 
-    const name = formData.get('name') as string;
-    if (!name?.trim()) {
-      newErrors.name = 'Server name is required';
-    }
-
-    if (Object.values(MCPServerName).includes(name as MCPServerName)) {
-      newErrors.name = 'Server name is already in use';
-    }
+  const defaultValues: FormValues = {
+    name: initialData?.name || '',
+    description: initialData?.description || '',
+    type: (initialData?.type || 'stdio') as 'stdio' | 'sse',
+    status: (initialData?.status || 'activate') as 'activate' | 'disabled',
+    command: (initialData as StdioServerData)?.command || '',
+    args: (initialData as StdioServerData)?.args?.join('\n') || '',
+    env:
+      Object.entries((initialData as StdioServerData)?.env || {})
+        .map(([key, value]) => `${key}=${value}`)
+        .join('\n') || '',
+    url: (initialData as SSEServerData)?.url || '',
+    authorization:
+      (initialData as SSEServerData)?.headers?.['Authorization']?.replace(
+        'Bearer ',
+        '',
+      ) || '',
+  };
 
-    if (serverType === 'stdio') {
-      const command = formData.get('command') as string;
-      if (!command?.trim()) {
-        newErrors.command = 'Command is required';
-      }
-    } else {
-      const url = formData.get('url') as string;
-      if (!url?.trim()) {
-        newErrors.url = 'URL is required';
-      } else if (!url.startsWith('http://') && !url.startsWith('https://')) {
-        newErrors.url = 'Please enter a valid URL';
-      }
-    }
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormValues>({
+    defaultValues,
+  });
 
-    console.log('newErrors', newErrors);
-    setErrors(newErrors);
-    return Object.keys(newErrors).length === 0;
-  };
+  const serverType = useWatch({
+    control,
+    name: 'type',
+  });
 
-  const onSubmitHandler = async (e: React.FormEvent<HTMLFormElement>) => {
-    console.log('onSubmitHandler', e);
-    e.preventDefault();
+  const status = useWatch({
+    control,
+    name: 'status',
+  });
 
+  const onSubmitHandler = async (data: FormValues) => {
     try {
       setIsLoading(true);
-      const formData = new FormData(e.currentTarget);
-
-      if (!validateForm(formData)) {
-        return;
-      }
 
       const baseData = {
```

Comment:
can use FormData like this:
```
const formData = new FormData(e.currentTarget);
const data = Object.fromEntries(formData.entries());

process({ ...data, id: uuidv4() });
```

Reply:
"handleSubmit" has already processed FormData into data